### DISCUSSION TITLE
Potential way to facilitate payment: Cloud vendor marketplaces

### DISCUSSION THREAD

**Discussion by choldgraf** [2i2c-org/team-compass#155] [Ideas]
# Background

In recent conversations with communities, we've realized that one pain-point is going to be administrative processing time and complexity. This is for a few reasons:

1. Universities can be difficult to work with, particularly for "new businesses" that aren't already in their system.
2. Universities often have pre-existing deals with cloud providers, meaning they'd like to pay the cloud provider directly
3. 2i2c has semi-limited payment processing capacity (e.g., it doesn't have a sophisticated credit card processing system)

# One option to explore

I had a conversation with Karan Bhatia, who works with the GCP research team. He said that vendors can apply for status on the GCP marketplace. If they do this, then there are a few benefits:

1. You have a negotiated rate with GCP that is often better than market rates
2. GCP sales folks will plug your service as a part of their marketplace offerings
3. The bill that people get is just a "regular" GCP bill with a line-item about the service, there is no vendor-specific payment
4. You then split that bill with Google according to the agreement

If we used something like this, it could be an easy way to rely on Google (or some other cloud provider, I assume this basic process exists with many) rather than putting that work on CSS. This might reduce the administrative time and complexity, and could also help us process different kinds of payments than straight-up invoicing (since we'd piggy back on Google's payment infrastructure).

Some docs about GCP billing process: https://cloud.google.com/marketplace/docs/understanding-billing

Curious what others think about this - if it is worth exploring etc.

ps: Karan emphasized that there no assumption of exclusivity here - you can put your service/product on any vendor marketplace at the same time, generally speaking.

**Comment by damianavila** [comment]
> Curious what others think about this - if it is worth exploring etc.

I think, at least, it is worth exploring... From a quick search, I see marketplaces for GCP, AWS, and Azure... so, not sure if they all behave closely but the concept is there and maybe we can take advantage of that!

**Reply by choldgraf** [reply]
Yeah - I will see if I can get more information from Karan, he said he'd be happy to put us in touch with the GCP sales folks. The one challenge here is that he said that getting on the marketplace also often comes with a bump in people asking for your service. This is not a bad thing in itself, but we should make sure we understand what to expect, and that we wouldn't get overloaded.

**Comment by rabernat** [comment]
From the point of view of a PI with existing funded projects for cloud computing, big :+1: to this idea. Being able to pay 2i2c with cloud credits would be HUGE! This also came up at the latest Pangeo Cloud Ops WG meeting. Rich Signell made the point that a big way to differentiate QHub from 2i2c was that you can pay for QHub completely with cloud credits.

**Reply by choldgraf** [reply]
@rabernat is QHub a marketplace product? I didn't realize they were treating it as a SaaS thing. I thought it was just a technology that they would deploy internally as a part of contracts, but it sounds like my info is outdated

**Reply by damianavila** [reply]
That's interesting, I had the same impression as @choldgraf...

**Reply by rabernat** [reply]
No, they are not a marketplace product. You run QHub on your own cloud billing account. But the point is it works if you only have cloud credits, not actual dollars to pay anyone.

Typically researchers will get a grant from AWS, Google, or MS for cloud research credits. They can spend them to run Qhub on their own account. But they can't spend them on 2i2c because we need to get paid actual dollars. Becoming a marketplace product would remove this distinction.

**Reply by choldgraf** [reply]
Ah I see - so qhub only costs credits in the sense that, if you ran any cloud infrastructure by yourself, it would only cost you cloud credits. (Similarly to if somebody set up their own custom pangeo-style hub)

Good point about the credits being more common than $$$ in academic research
